### PR TITLE
add support for specifying protocols on websocket handshake

### PR DESCRIPTION
**Pull request by jairinhohw**
issue #821

### CODE REVIEW

File: actix-web-actors/src/ws.rs [hunk 1]
```diff
@@ -31,7 +31,24 @@ where
...
+    let mut res = handshake(req, &vec![])?;
+    Ok(res.streaming(WebsocketContext::create(actor, stream)))
+}
+
+/// Do websocket handshake and start ws actor.
+///
+/// `protocols` is a sequence of known protocols.
+pub fn start_with_protocols<A, T>(
+    actor: A,
+    protocols: &Vec<&str>,
```
Comment (fafhrd91) [inline actix-web-actors/src/ws.rs]: let's change this to `&[str]`

File: actix-web-actors/src/ws.rs [hunk 2]
```diff
@@ -40,10 +57,13 @@ where
...
-// /// `protocols` is a sequence of known protocols. On successful handshake,
-// /// the returned response headers contain the first protocol in this list
-// /// which the server also knows.
-pub fn handshake(req: &HttpRequest) -> Result<HttpResponseBuilder, HandshakeError> {
+/// `protocols` is a sequence of known protocols. On successful handshake,
+/// the returned response headers contain the first protocol in this list
+/// which the server also knows.
+pub fn handshake(
+    req: &HttpRequest,
+    protocols: &Vec<&str>,
```
Comment (fafhrd91) [inline actix-web-actors/src/ws.rs]: this is breaking change, let's use new function

Comment (fafhrd91) [reply actix-web-actors/src/ws.rs]: also, should this be `Option<&[str]>`?

**Review by fafhrd91 — CHANGES_REQUESTED** [review]

File: actix-web-actors/src/ws.rs (hunk 2 above)
Comment (jairinhohw) [reply actix-web-actors/src/ws.rs]: it would make no sense to pass a `Some([])`, so i though to pass the empty vec to not use a protocol

**Comment by fafhrd91** [comment]
thanks

**Pull request merged**